avm/res/network/network-manager/security-admin-configuration/rule-collection/main.bicep

```diff
@@ -35,28 +35,28 @@ resource ruleCollection 'Microsoft.Network/networkManagers/securityAdminConfigur
   name: name
   parent: networkManager::securityAdminConfiguration
   properties: {
-    description: description
+    description: description ?? ''
     appliesToGroups: appliesToGroups
   }
 }
 
 module securityAdminConfigurations_rules 'rule/main.bicep' = [
-  for (rule, index) in rules: {
+  for (rule, index) in rules ?? []: {
     name: '${uniqueString(deployment().name)}-RuleCollections-Rules-${index}'
     params: {
       networkManagerName: networkManager.name
       securityAdminConfigurationName: securityAdminConfigurationName
       ruleCollectionName: ruleCollection.name
       name: rule.name
       access: rule.access
-      description: contains(rule, 'description') ? rule.description : ''
-      destinationPortRanges: contains(rule, 'destinationPortRanges') ? rule.destinationPortRanges : []
-      destinations: contains(rule, 'destinations') ? rule.destinations : []
+      description: rule.?description ?? ''
+      destinationPortRanges: rule.?destinationPortRanges ?? []
```

Comment:
Just a suggestion (so feel free to just resolve), but you could actually leave the else case out when referencing the child module. 

Example
- If in the child module `param destinationPortRanges` has a default in the child, and this line here is just `destinationPortRanges: rule.?destinationPortRanges`, then it will automatically pick the default in the child
- If in the child module `param destinationPortRanges` has no default but is nullable like in your case, then no value is passed through (which I guess also is fine)